chemreg/compound/tests/test_filters.py

```diff
@@ -1,24 +1,42 @@
 import pytest
 
 
+@pytest.mark.parametrize("search_type", ["V3000", "V2000", "SMILES"])
 @pytest.mark.django_db
-def test_defined_compound_filters(defined_compound_factory, user, client):
+def test_defined_compound_filtors(
+    search_type,
+    defined_compound_factory,
+    defined_compound_v2000_factory,
+    defined_compound_smiles_factory,
+    user,
+    client,
+):
+    def clean_molfile(mol):
+        return (
+            mol.replace("-INDIGO-", "-CHEMREG-").replace(" ", "+").replace("\n", "%0A")
+        )
+
+    factory_dict = {
+        "V3000": ("molfile_v3000", defined_compound_factory),
+        "V2000": ("molfile_v2000", defined_compound_v2000_factory),
+        "SMILES": ("smiles", defined_compound_smiles_factory),
+    }
     client.force_authenticate(user=user)
-    compounds = defined_compound_factory.create_batch(2)
+    field, factory = factory_dict.get(search_type)
+    factory.create()
```

Comment:
This actually looks like it might be to create seed data for results.  In that case perhaps a comment explaining it makes more sense?
```suggestion
    # Add additional results to be filtered out.
    factory.create()
```

Reply:
good idea, and you are correct that it's just to filter out!